src/WxWork/Provider.php

```diff
@@ -0,0 +1,155 @@
+        $this->clientSecret = $config['client_secret'];
+        $this->redirectUrl = $config['redirect'];
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getAuthUrl($state)
+    {
+        $isWxWork = Str::is('*wxwork*', request()->userAgent());
+
+        if ($isWxWork) {
+            //企业微信内部静默授权
+            return $this->buildAuthUrlFromBase('https://open.weixin.qq.com/connect/oauth2/authorize', $state).'#wechat_redirect';
+        } else {
+            //非企业微信客户端 使用扫码登录
+            return $this->buildAuthUrlFromBase('https://open.work.weixin.qq.com/wwopen/sso/qrConnect', $state);
+        }
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function buildAuthUrlFromBase($url, $state)
+    {
+        $query = http_build_query($this->getCodeFields($state), '', '&', $this->encodingType);
+
+        return $url.'?'.$query;
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getCodeFields($state = null)
+    {
+        return [
+            'appid'         => $this->corpid,
+            'agentid'       => $this->clientId,
+            'redirect_uri'  => $this->redirectUrl,
+            'state'         => $state,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://qyapi.weixin.qq.com/cgi-bin/user/getuserinfo';
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get('https://qyapi.weixin.qq.com/cgi-bin/user/get', [
+            'query' => [
+                'access_token' => $token,
+                'userid'       => $this->userid,
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'        => Arr::get($user, 'userid'),
+            'nickname'  => $user['alias'] ?? $user['name'],
+            'name'      => $user['name'],
+            'avatar'    => $user['avatar'],
+            'email'     => $user['email'],
+            'mobile'    => $user['mobile'],
+            'gender'    => $user['gender'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    public function getAccessTokenResponse($code)
+    {
+        $response = $this->getHttpClient()->get($this->getTokenUrl(), [
+            'query' => [
+                'access_token'  => $this->getAccessToken()['access_token'],
+                'code'          => $code,
+            ],
+        ]);
+        $this->userid = json_decode($response->getBody(), true)['UserId'];
+        $this->credentialsResponseBody = $this->getAccessToken();
+
+        return $this->credentialsResponseBody;
+    }
+
+    /**
+     * get access token.
+     *
+     * @return void
+     */
+    protected function getAccessToken()
```

Comment:
Why is this being cached?

Reply:
AccessToken has a relatively long validity period,
Wxwork has limited frequency of obtaining AccessToken.